### PR TITLE
Add some handles to allow more customizations

### PR DESCRIPTION
**Pull request by ttedeschi**
- Add a way for tweaking "manually" the position of CMS and extra text
- Make `tdrCanvas` and `tdrDiCanvas` functions return pad frames, too

### CODE REVIEW

File: tdrstyle.py [hunk 1]
```diff
@@ -379,7 +381,7 @@ def tdrCanvas(canvName, x_min, x_max, y_min, y_max, nameXaxis, nameYaxis, square
     UpdatePad(canv)
     canv.RedrawAxis()
     canv.GetFrame().Draw()
-    return canv
+    return canv, h
```
Comment (anmalara) [inline tdrstyle.py:384]: why is this needed?
There is the function GettdrCanvasHist to retrieve the hist, for the few cases when it's needed.
Generally, what's the need to handle the frame directly?

Also, if you want to modify this function, the example has to be updated too.

File: tdrstyle.py [hunk 2]
```diff
@@ -25,6 +25,8 @@ def SetLumi(lumi, round_lumi=False):
 
 cmsText     = 'CMS'
 extraText   = 'Preliminary'
+cmsTextMove = 0
+extraTextMove = 0
```
Comment (anmalara) [inline tdrstyle.py:29]: I though that the idea was to keep the customisations to a minimum.
What's the idea behind these changes?

Comment (ttedeschi) [reply tdrstyle.py:29]: The idea behind this is to try and implement the pyROOT conterpart of the snippets shown here https://cms-analysis.docs.cern.ch/guidelines/plotting/general/ for mplhep (I'm implementing them here https://gitlab.cern.ch/cms-analysis/docs/-/merge_requests/43)
You're right anyway, we can use `GettdrCanvasHist` without modifying the return of the functions, thanks for the suggestion, let me try it and modify the PR

Comment (anmalara) [reply tdrstyle.py:29]: ok I see.
Have you tried with cmsText=ScaleText("#times 10^{5}", scale = 0.15)+" CMS"? Or something similar? in this way the shift wouldn't need to be checked by try-and-error.

Comment (ttedeschi) [reply tdrstyle.py:29]: How do you suggest we can handle the scientific notation factor like this (scaling the text) when the scientific notation is produced automatically via `hdf.GetYaxis().SetMaxDigits(1)` and not hardcoded? 
By the way if you do something like this `TDR.cmsText=TDR.ScaleText("#times 10^{5}", scale = 0.15)+" CMS"` it would end up in bold.. maybe in this case it is better to add a variable like `TDR.prefixText`?

Comment (andrzejnovak) [reply tdrstyle.py:29]: Does this option need to be provided by the code as opposed to left to the user? The user can already just choose in-frame label when using a sci scale on top.

Comment (anmalara) [reply tdrstyle.py:29]: I agree, it seems too customisable. Also, I played a bit and if we shift the CMS tag, it would screw up the relative position with the extra text. And it's not easy to catch all cases with the spaces needed for the y-axis label.

For the few cases in which the scientific notation is needed and the CMS tag cannot be used inside the frame, people could use something like this:

```
TDR.GettdrCanvasHist(canv).GetYaxis().SetMaxDigits(1)
ROOT.TGaxis.SetExponentOffset(-0.07,0.015,"y")
```
to obtain:
Screenshot 2023-11-02 at 09 18 18

Comment (ttedeschi) [reply tdrstyle.py:29]: I agree, too. I was just trying to match the second "case" in https://gitlab.cern.ch/ttedesch/docs/-/blob/master/docs/guidelines/plotting/general.md?ref_type=heads#user-content-scientific-notation in which it is shown explicitly how to move the CMS label "manually" in mplhep
So if you agree, @anmalara @andrzejnovak we can delete that second "case" directly from the docs

For the snippet of code you suggest @anmalara , it refers to the first "case" in that section. That very same piece of code was already in the MR I prepared https://gitlab.cern.ch/cms-analysis/docs/-/merge_requests/43/diffs#88dc78f23b273cf12108a587e01e3a5d19df3c5b_124_126

Comment (anmalara) [reply tdrstyle.py:29]: ah good. I haven't seen that. Sounds good to me. Thanks :-)

**Comment by ttedeschi** [comment]
closing as no modification to original code is required after discussion
